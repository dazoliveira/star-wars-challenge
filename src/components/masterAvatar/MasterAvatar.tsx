import React from "react";
import { Avatar, Grid, Typography } from "@material-ui/core";
import { useStyles } from "./styles";

export type MasterAvatarProps = {
    variant: boolean
    isMobile: boolean
    avatarSrc: string
    text: string
    avatarName?: string | null
}

export default function MasterAvatar({ variant, isMobile, avatarSrc, avatarName, text }: MasterAvatarProps) {

    const classes = useStyles({ variant, isMobile })
    
    return (
        <Grid 
          container
          item
          direction='column'
          justifyContent="center"
          alignItems="center"
          xs={12}
        >          
        <Avatar 
          alt={avatarName || 'master-avatar'}
          src={avatarSrc}
          className={classes.avatar} 
        />            
        <Typography 
          variant='h3'
          component='p'
          color='secondary'
          align='center'
          className={classes.name}
        >
          {text}<span>{avatarName || '. . .'}</span>
        </Typography>
      </Grid>   
    )
}