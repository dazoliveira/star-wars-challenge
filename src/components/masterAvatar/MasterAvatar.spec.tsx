import { screen } from '@testing-library/react'
import { renderWithTheme } from "utils/tests/helpers";
import MasterAvatar, { MasterAvatarProps } from './MasterAvatar';
import imgTest from '../../features/jediVsDarkseid/imgs/luke-skywalker.png'

const props: MasterAvatarProps = {
    variant: true,
    isMobile: false,
    avatarSrc: imgTest,
    text: 'Your master is',
    avatarName: 'Luke Skywalker'
}

describe('<MasterAvatar />', () => {
    it('should render master avatar', () => {
        const { container } = renderWithTheme(<MasterAvatar {...props} />)
        expect(screen.getByText(/Your master is/i)).toBeInTheDocument()
        expect(screen.getByText(/Luke Skywalker/i)).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()

    })

    it('should render Luck Skywalker image', async () => {
       renderWithTheme(<MasterAvatar {...props} />)

       const displayedImage = document.querySelector("img") as HTMLImageElement;
       expect(displayedImage.src).toContain("luke-skywalker.png");
    })
});