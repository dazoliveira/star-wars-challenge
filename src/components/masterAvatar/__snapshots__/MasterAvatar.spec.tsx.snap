// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MasterAvatar /> should render master avatar 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center MuiGrid-grid-xs-12"
>
  <div
    class="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-1 makeStyles-avatar-3"
  >
    <img
      alt="Luke Skywalker"
      class="MuiAvatar-img"
      src="luke-skywalker.png"
    />
  </div>
  <p
    class="MuiTypography-root makeStyles-name-2 makeStyles-name-4 MuiTypography-h3 MuiTypography-colorSecondary MuiTypography-alignCenter"
  >
    Your master is
    <span>
      Luke Skywalker
    </span>
  </p>
</div>
`;
